xbmc/guilib/GUIControlFactory.cpp

```diff
@@ -342,7 +342,11 @@ bool CGUIControlFactory::GetTexture(const TiXmlNode* pRootNode, const char* strT
   if (!pNode) return false;
   const char *border = pNode->Attribute("border");
   if (border)
+  {
     GetRectFromString(border, image.border);
+    const char *borderinfill = pNode->Attribute("infill");
+    image.infill = !(borderinfill && StringUtils::CompareNoCase(borderinfill, "false") == 0);
```

Comment:
Better avoid the global negation to increase readability, the way it's written is confusing

Reply:
Like:
`if (borderinfill && StringUtils::CompareNoCase(borderinfill, "false") == 0)
  image.infill = false;`

...and it defaults to true implicitly?

Reply:
!(A && B) = (!A || !B) so it should be this:
`image.infill = !borderinfill || StringUtils::CompareNoCase(borderinfill, "false") != 0;`

you could also use `EqualsNoCase` instead `CompareNoCase`

Reply:
Settled for `image.infill = !(borderinfill && StringUtils::CompareNoCase(borderinfill, "false") == 0);`